qiskit_experiments/framework/base_analysis.py

```diff
@@ -154,6 +154,11 @@ def run(
             will be returned containing only the new analysis results and figures.
             This data can then be saved as its own experiment to a database service.
         """
+        experiment_data.add_artifacts(
+            ArtifactData(name="analysis_config", data=self.config()),
+            overwrite_name=True,
+            overwrite_id=True,
+        )
```

Comment:
If `replace_results` is `False` then we create a copy of the `ExperimentData` object. Both the original object and its copy will now contain the analysis config artifact, although the original one will not contain the corresponding analysis result.

Reply:
I'm also concerned about the old analysis config that's stored in the experiment data and is now overridden. The concern applies for the two possible values of `replace_results`.